File: Algorithms/Searching/BinarySearch.java
class BinarySearch {
    public static void main(String[] args) {
        int[] arr = { 1, 2, 3, 4, 5 };
        System.out.println(binarySearch(arr, 0, arr.length - 1, 2));
    }

    static int binarySearch(int[] arr, int l, int r, int item) {
        // item in mid
        int mid = (l + r) / 2;
        if (arr[mid] == item)
            return mid;

        // item in left or right
        if (l < r) {
            if (arr[mid] < item)
                return binarySearch(arr, mid + 1, r, item);
            else
                return binarySearch(arr, l, mid - 1, item);
        }
        return -1;
    }
}

File: Algorithms/Searching/BinarySearch.js
const arr = [1, 2, 3, 4, 5];
const item = 5;

function binarySearch(arr, left, right, item) {
  if (left <= right) {
    let mid = Math.floor((left + right) / 2);
    let midItem = arr[mid];

    if (midItem === item) {
      return console.log("found at index:", mid);
    } else if (item < midItem) {
      return binarySearch(arr, left, mid - 1, item);
    } else if (item > midItem) {
      return binarySearch(arr, mid + 1, right, item);
    }

    return console.log("not found!");
  }
}

binarySearch(arr, 0, arr.length, item);
